lib/devcycle-ruby-server-sdk/api/devcycle_api.rb

```diff
@@ -36,13 +36,37 @@ def initialize(sdkKey, dvc_options = DVCOptions.new, &initialize_callback)
       nil
     end
 
+    def close
+      if @dvc_options.enable_cloud_bucketing
+        @logger.info("Cloud Bucketing does not require closing.")
+        return
+      end
+      if @localbucketing != nil
```

Comment:
why not use the `local_bucketing_initialized` method for the check here? might save you from having the nested ifs

Reply:
Because they're two different states; the client can be initialized; and not initialized - and we need to handle the cases independently

Reply:
wouldn't something like this work?
```
while !local_bucketing_initialized?
    sleep(0.5)
end
@localbucketing.close
@logger.info("Closed DevCycle Local Bucketing Engine.")
```
cause if local bucketing isn't initialized, then you'll just wait in the while loop until it is. Otherwise just close

Reply:
If the client isn't initialized in that case it'll block forever.

Reply:
ah okay, makes sense